service/k8sclient/k8sclient.go

```diff
@@ -1209,21 +1230,38 @@ func (c *K8sClient) CheckOperators(ctx context.Context) (*Operators, error) {
 	}, nil
 }
 
-func (c *K8sClient) checkOperatorStatus(installedVersions []string, expectedAPIVersion string) (operator OperatorStatus) {
+func (c *K8sClient) checkOperatorStatus(installedVersions []string, expectedAPIVersion string) (operator Operator) {
 	apiNamespace := strings.Split(expectedAPIVersion, "/")[0]
-	installed := false
-	for _, version := range installedVersions {
-		switch {
-		case version == expectedAPIVersion:
-			return OperatorStatusOK
-		case strings.HasPrefix(version, apiNamespace):
-			installed = true
+	operator.Status = OperatorStatusNotInstalled
+	lastVersion, _ := version.NewVersion("v0.0.1")
+	for _, apiVersion := range installedVersions {
+		if !strings.HasPrefix(apiVersion, apiNamespace) {
+			continue
 		}
+		if apiVersion == expectedAPIVersion {
+			operator.Status = OperatorStatusOK
+		}
+		if operator.Status == OperatorStatusNotInstalled {
+			operator.Status = OperatorStatusUnsupported
+		}
+		v := strings.Split(apiVersion, "/")[1]
+
+		versionParts := strings.Split(v, "-")
+		if len(versionParts) != 3 {
+			continue
+		}
+		v = strings.Join(versionParts, ".")
+		newVersion, err := version.NewVersion(v)
+		if err != nil {
+			c.l.Warn("can't parse version %s: %s", v, err)
+			continue
+		}
+		if newVersion.LessThanOrEqual(lastVersion) {
+			continue
+		}
+		operator.Version = newVersion.String()
```

Comment:
@BupycHuk I am not sure what this func does. Don't we every time overwrite `operator.Version` when multiple versions of CRD is installed?

Reply:
when user installs 1.7.0 it includes API versions from 1-0-0 to 1-7-0, so we have to select the one with highest value to figure out which version is installed right now.

Reply:
Well then assignment `lastVersion = newVersion.String()` is missing inside the loop, right? Otherwise the lastVersion stays on `v0.0.1`

Reply:
It would be nice to have a docstring with an example of this function. So in future, we can easily recall its purpose.

Reply:
yes, that's correct.

Reply:
@askomorokhov @jprukner done